generator/common/beatgen/setup/setup.go

```diff
@@ -18,111 +18,132 @@
 package setup
 
 import (
+	"bufio"
 	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
 
 	"github.com/magefile/mage/sh"
 	"github.com/pkg/errors"
 
 	devtools "github.com/elastic/beats/dev-tools/mage"
+	"github.com/elastic/beats/dev-tools/mage/gotool"
 )
 
-// RunSetup runs any remaining setup commands after the vendor directory has been setup
-func RunSetup() error {
-	vendorPath := "./vendor/github.com/"
-
-	//Copy mage stuff
-	toMkdir := filepath.Join(vendorPath, "magefile")
-	err := os.MkdirAll(toMkdir, 0755)
-	if err != nil {
-		return errors.Wrapf(err, "error making mage directory at %s", toMkdir)
+var (
+	makefileDeps = []string{
+		filepath.Join("dev-tools"),
+		filepath.Join("libbeat", "scripts"),
+		filepath.Join("libbeat", "tests", "system", "requirements.txt"),
+		filepath.Join("metricbeat", "Makefile"),
+		filepath.Join("metricbeat", "_meta", "common.reference.yml"),
+		filepath.Join("metricbeat", "_meta", "common.yml"),
```

Comment:
Are you copying these out of `beats/metricbeat` or somewhere in `dev-tools` ? I feel like the generator shouldn't be depending on metricbeat configs. The generator should probably use its own config files.

Reply:
These files are from `beats/metricbeat`. They are various dependencies of configuration and fields generators. If someone is generating a new Metricbeat metric set, he/she needs these in order to get a working config and assets file. It has been a dependency previously. I don't see why we should change it now.

Previously, the usage of these files was not "visible" because the whole Beats repository was copied to vendor. Now only Go source files are moved to the folder using `go mod vendor`. So now moving these files has to be done manually.

Reply:
Some observations about this:
* Are we copying these files also for new beats of `beat` type?
* How are we going to handle multiple versions for this?
  * If a new non-Go dependency is added, we should detect this on tests (I guess we will do it with `make test test-package`)
  * If someone selects a `beats_revision` with different dependencies it should still work.
  * What would be the procedure to update a beat to a new version with different dependencies?
* Could we take advantage of something like https://github.com/goware/modvendor?

Reply:
* Yes. I am OK with skipping metricbeat files if the type is not metricbeat.
* What do you mean by multiple versions? Versions where these files are not available?
* It's not necessary to use an external tool. It is already implemented for Beats, as we have a few deps with C source files.

Reply:
With multiple versions I mean that the generator is run from one version, and the user can chose any value for `beats_revision`. The user could select versions that need a different set of files. How are we going to handle that?

Reply:
I will document that case.